source/lmp/pair_deepmd.cpp

```diff
@@ -344,11 +344,17 @@ PairDeepMD::PairDeepMD(LAMMPS *lmp)
   if (lmp->citeme) {
     lmp->citeme->add(cite_user_deepmd_package);
   }
-  if (strcmp(update->unit_style, "metal") != 0) {
-    error->all(
-        FLERR,
-        "Pair deepmd requires metal unit, please set it by \"units metal\"");
+  int unit_convert;
+  if (strcmp(update->unit_style, "metal") == 0) {
+    unit_convert = utils::NOCONVERT;
+  } else if (strcmp(update->unit_style, "real") == 0) {
+    unit_convert = utils::METAL2REAL;
+  } else {
+    error->all(FLERR,
+               "Pair deepmd requires metal or real unit, please set it by "
+               "\"units metal\" or \"units real\"");
   }
+  conversion_factor = utils::get_conversion_factor(utils::ENERGY, unit_convert);
```

Comment:
It looks OK. 

We need only one `conversion_factor`, because the length unit are the same and we only need to convert energy. It may be a better idea to express the fact in the name of the factor, e.g. change it to `ener_unit_cvt_factor`